nifi-commons/nifi-json-schema-utils/pom.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.nifi</groupId>
+        <artifactId>nifi-commons</artifactId>
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>nifi-json-schema-utils</artifactId>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+   <dependencies>
+       <dependency>
+           <groupId>org.apache.nifi</groupId>
+           <artifactId>nifi-api</artifactId>
+       </dependency>
+       <dependency>
+           <groupId>com.networknt</groupId>
+           <artifactId>json-schema-validator</artifactId>
+           <version>1.0.86</version>
+       </dependency>
```

Comment:
Since this is an API module, this dependency should be removed. The `SchemaVersion` should not have a direct reference to the library class. Although there may be some small duplication, do not recommend placing the PropertyDescriptor in a shared module for now.

Reply:
Addressing he first point. The dependency is needed for `SchemaVersion` enum as it used to be in ValidateJson and it uses `SpecVersion.VersionFlag` which is part of the dependency.
As for your second point, where should I put the `PropertyDescriptor`  as I created a common one `SCHEMA_VERSION` used in `ValidateJson` and `InMemoryJsonSchemaRegistry`

Reply:
Thanks for the quick reply. Instead of having the enum include a reference to the `VersionFlag`, the Processor should have a translation method to determine the VersionFlag from the `SchemaVersion` value.

In this particular case, I recommend simply duplicating the Property Descriptor definition in both components.

Reply:
So this translation method will be needed in `ValidateJson` and `InMemoryJsonSchemaRegistry`. Where could i place this common method so it is not duplicated?

Reply:
Does the Registry implementation need the translation method? It seems like it is just storing the text and version information, so the translation method could be in ValidateJson.

Reply:
It uses the translation to validate the schema in the customValidate to ensure one enters a valid schema per the version.

Reply:
Reviewing the [VersionFlag](https://github.com/networknt/json-schema-validator/blob/master/src/main/java/com/networknt/schema/SpecVersion.java#L24) code, it looks like the `id` field is the URI of the schema version. Adding the URI value to the `SchemaVersion` seems like a good approach, and then it is simply a matter of calling `VersionFlag.fromId()`, so no shared utility should be needed.

Reply:
That is a great idea. I will look to implement that. As regards to defining a common `PropertyDescriptor`, I followed the practice I saw in` nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/schema/access/SchemaAccessUtils.java` where a handful of `PropertyDescriptor` variables are defined. In addition I wanted to avoid what I saw in #7890 where there was a cut and paste of multiple  `PropertyDescriptor` variables in two separate JOLT processors and as a result the same bug had to be fixed in both places.

Reply:
Thanks for highlighting the positives and negatives of a shared PropertyDescriptor. For one property, the value seems marginal, but it could have some usefulness. For shared PropertyDescriptors, I recommend an approach like the [KafkaClientComponent](https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-shared/src/main/java/org/apache/nifi/kafka/shared/component/KafkaClientComponent.java) which is an interface that consists exclusively of Property Descriptors. At that point, the question is where this should be placed, as it should not be considered part of the API per se. Following the above approach, a `json-schema-shared` library seems the most logical.